### PR TITLE
Add more date ticks

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

Add use-x-tick-count hook to determine how many x-ticks need to be rendered at a given screen size
_This is a spike, meant to be discussed. The implementation doesn't necessarily have to end up being merged. But, I think this works pretty well._

image

image

image

## Motivation

Design request.

## Detailed design

I added a hook called `useXTickConfiguration`. In this hook, based on the type of values (date or datespan), the length of the value collection and the chart size and pre-configured number of ticks and the date format are chosen.
This hook also uses the new hook `useChartBreakpoints` which is similar to the `useBreakpoints` hook only it uses the chart width to determine the breakpoint values.

I've fiddled with different amounts and the settings in this PR simply looked the best. I'm sure a completely generic approach that measures every tick and container can be conjured up, but this pretty much works.

The date format is passed down to the Axes component where it is used by its `formatXAxis` method.

Currently, the configuration for the tick count / screen size is hardcoded inside the `useXTickCount` hook, this is because it's only used by the time-series-chart for now. Obviously, if more than one config is needed we can choose to pass it in as an argument.

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import {
+  assert,
+  isDateSeries,
+  isDateSpanSeries,
+  TimestampedValue,
+} from '@corona-dashboard/common';
+import { useBreakpoints } from '~/utils/use-breakpoints';
+
+interface XTickConfiguration {
```
Comment (VWSCoronaDashboard8) [inline packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts:9]: What does X stand for?

Comment (VWSCoronaDashboard1) [reply packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts:9]: X Axis

Comment (VWSCoronaDashboard10) [reply packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts:9]:
```suggestion
interface XAxisTickConfiguration {
```

File: packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  date: {
+    long: { count: number; format?: any };
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts:11]: oh, the format should be removed completely, that was just a remnant of some older code

File: packages/app/src/components/time-series-chart/components/axes.tsx [hunk 3]
```diff
@@ -115,16 +117,13 @@ export const Axes = memo(function Axes({
...
-      const startYear = createDate(startUnix).getFullYear();
-      const endYear = createDate(endUnix).getFullYear();
-
-      const isMultipleYearSpan = startYear !== endYear;
-
-      return isMultipleYearSpan && [startUnix, endUnix].includes(date_unix)
-        ? formatDateFromSeconds(date_unix, 'axis-with-year')
-        : formatDateFromSeconds(date_unix, 'axis');
+      if (xTickValues.length < 6) {
+        return formatDateFromSeconds(date_unix, 'axis-with-year-long');
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply packages/app/src/components/time-series-chart/components/axes.tsx:131]: I'm using them for the full screen mode now

File: packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+export function useXTickCount<T extends TimestampedValue>(values: T[]) {
+  const isDateValues = isDateSeries(values);
+  const isDateSpanValues = isDateSpanSeries(values);
+  assert(isDateValues || isDateSpanValues, 'Unknown date series encountered');
+
+  const type = isDateValues ? 'date' : 'span';
+  const period = values.length < 36 ? 'short' : 'long';
+
+  const breakpoints = useBreakpoints(true);
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply packages/app/src/components/time-series-chart/logic/use-x-tick-count.ts]: I've created the `useChartBreakpoints` hook for this. The widths I determined by resizing the app to the different screen sizes in dev tools and determining the width of the chart there.

File: packages/app/src/components/time-series-chart/components/axes.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply packages/app/src/components/time-series-chart/components/axes.tsx:131]: image

[the remark this answers is not in the document]
Comment (VWSCoronaDashboard10) [reply packages/app/src/components/time-series-chart/components/axes.tsx]: Yes I understand but it looks kind of off to repeat the year so often

**Pull request closed without merging**